cips/cip-shwap_protocol.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Shwap Protocol 
+description: Shwap - the new extensible sampling protocol 
+author: Hlib Kanunnikov (@Wondertan)
+discussions-to: https://forum.celestia.org/t/cip-shwap-protocol/1551
+status: Draft
+type: Standards Track
+category: Data Availability, Networking
+created: 2024-02-02
+---
+
+## Abstract
+
+This document describes a high level overview of the Shwap p2p protocol. Shwap provides scalable and extensible 
+framework for exchanging and swapping of shared data for Celestia's Data Availability network and beyond.
+
+## Motivation
+
+Current Data Availability Sampling (DAS) network protocol is inefficient. A _single_ sample operation takes O(log2(N)) network
+round-trips(where N is the square size). This is not practical and does not scale for theoretically unlimited data square
+Celestia network enables. The main motive here is a protocol with O(1) round-trip for _multiple_ samples, preserving 
+the assumption of having a 1/N honest peers connected.
+
+Initially, the Bitswap and IPLD were adopted as a base for the DA network protocols, including DAS, 
+block synchronization (BS) and blob/namespace data retrieval (ND). They gave battle tested protocols and tooling with 
```

Comment:
[grammar]


Initially, Bitswap and IPLD were adopted as the basis for the DA network protocols, including DAS, block synchronization (BS), and blob/namespace data retrieval (ND).